### PR TITLE
Fixed the 'Bad Credentials' translation problem

### PR DESCRIPTION
**Pull request by ejkun**
As related in this [issue](https://github.com/sonata-project/SonataUserBundle/issues/621) there was a problem with the error message in the login page, that was not being translated. That was a error with FOSUserBundle, that apparently will be fixed in the 2.0 version, but since there is no prediction to when it will be ready. I made some changes by myself based on the dev-master of FOSUserBundle.

I changed both security controllers and the templates for the login page to look more like the latest version of FOSUserBundle. Now the error message is translated with the 'security'. Also took out the Bad Credentials entry in one of the translation files since there is no need for it.

### CODE REVIEW

**Comment by OskarStark** [comment]
nice job @ejkun 

can you please add a test case for this implementation?

**Comment by ejkun** [comment]
@OskarStark , I'm not very good with tests. Which test should I do? Test the controller? Since there are no tests for the Controllers or the Views, I don't know where to start.

**Comment by ejkun** [comment]
I was checking the FOSUserBundle to see if they have some tests. Well, they don't. None of the controllers or views are tested. So I really don't know where to start.

**Comment by greg0ire** [comment]
@OskarStark I don't think asking for tests is fair in that case.

**Comment by greg0ire** [comment]
@ejkun , about your commit message, can you read and apply : https://github.com/Soullivaneuh/sonata-dev-kit/blob/doc-contributing/doc/CONTRIBUTING.md#the-commit-message (especially the fix about part).

Also, your commit message descriptions contains the word "also". There should probably be two commits. Use

```
git reset origin/master
git add --patch
```
to forget your commit and rebuild them block by block.

File: Resources/views/Admin/Security/login.html.twig [hunk 1]
```diff
@@ -30,7 +30,7 @@
             {% block sonata_user_login_form %}
                 {% block sonata_user_login_error %}
                     {% if error %}
-                        <div class="alert alert-danger">{{ error|trans({}, 'FOSUserBundle') }}</div>
+                        <div class="alert alert-danger alert-error">{{ error.messageKey|trans(error.messageData, 'security') }}</div>
```
Comment (greg0ire) [inline Resources/views/Admin/Security/login.html.twig]: This line is too long.

File: Resources/views/Security/base_login.html.twig [hunk 2]
```diff
@@ -25,7 +25,7 @@ file that was distributed with this source code.
 
                     {% block sonata_user_login_error %}
                         {% if error %}
-                            <div class="alert alert-danger alert-error">{{ error|trans({}, 'FOSUserBundle') }}</div>
+                            <div class="alert alert-danger alert-error">{{ error.messageKey|trans(error.messageData, 'security') }}</div>
```
Comment (greg0ire) [inline Resources/views/Security/base_login.html.twig]: This line is too long.

File: Controller/SecurityFOSUser1Controller.php [hunk 3]
```diff
@@ -37,6 +37,41 @@ public function loginAction()
...
+        }
+
+        if (!$error instanceof AuthenticationException) {
+            $error = null; // The value does not come from the security component.
+        }
+
+        // last username entered by the user
+        $lastUsername = (null === $session) ? '' : $session->get(SecurityContextInterface::LAST_USERNAME);
+
+        $csrfToken = $this->container->get('form.csrf_provider')->generateCsrfToken('authenticate');
```
Comment (greg0ire) [inline Controller/SecurityFOSUser1Controller.php]: Both these variables are used only once, as values for keys that have the same name. You could write directly :

```php
array(
    'last_username' => (null === $session) ? '' : $session->get(SecurityContextInterface::LAST_USERNAME);
    'csrf_token' => $this->container->get('form.csrf_provider')->generateCsrfToken('authenticate');
)
```

File: Controller/SecurityFOSUser1Controller.php [hunk 4]
```diff
@@ -37,6 +37,41 @@ public function loginAction()
             return new RedirectResponse($url);
         }
 
-        return parent::loginAction();
+        /*
+            Implementation of parent::loginAction
+            Needed for fixing the "Bad Credentails" translation problem
+            The code is a mix of current dev-master
```
Comment (greg0ire) [inline Controller/SecurityFOSUser1Controller.php]: "current" will lose its meaning ;)

File: Controller/SecurityFOSUser1Controller.php [hunk 5]
```diff
@@ -37,6 +37,41 @@ public function loginAction()
             return new RedirectResponse($url);
         }
 
-        return parent::loginAction();
+        /*
+            Implementation of parent::loginAction
+            Needed for fixing the "Bad Credentails" translation problem
```
Comment (greg0ire) [inline Controller/SecurityFOSUser1Controller.php]: Credentials

**Comment by ejkun** [comment]
ping @OskarStark

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**